independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java

```diff
@@ -62,7 +61,7 @@ private double calculateScore(OriginCombination s) {
         for (OriginWithPreference o : s.getCollectedOrigins()) {
             combinationScore += Math.pow(extOrigins.size(),
                     highestRegistryPreference + 1 - o.getPreference().registryPreference)
-                    * ((double) (Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
+                    * (((double) Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```

Comment:
```suggestion
                    * ((((double) Integer.MAX_VALUE) + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```

@aloubyansky I would extract this to a static method and make sure it works as expected with unit tests, this is a tricky computation.

Reply:
talking about this block 
```java
Math.pow(extOrigins.size(),
                    highestRegistryPreference + 1 - o.getPreference().registryPreference)
                    * (((double) Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE)
```

Reply:
The formula itself isn't particularly interesting to test in isolation. What counts is which platform and extension versions are suggested in different contexts, for that I do add tests.

Reply:
any formula is interesting to test in isolation IMHO to make sur the calculation is done right. Even more when there is casts and different types. For example, if registryPreference or platformPreference type changes, that could make it fail again, while with a strong typed test on
`computeScore(int, int, int, int)` (or whatever) make sure it is used right and valid in common cases.

This is just a suggestion.. I am not blocking the PR ;)